lib/AST/Type.cpp

```diff
@@ -2452,9 +2452,18 @@ Type ReplaceOpaqueTypesWithUnderlyingTypes::operator()(
   auto archetypeAndRoot = getArchetypeAndRootOpaqueArchetype(maybeOpaqueType);
   if (!archetypeAndRoot)
     return maybeOpaqueType;
-  
+
   auto archetype = archetypeAndRoot->first;
   auto opaqueRoot = archetypeAndRoot->second;
+
+  // Don't replace opaque types from resilient modules.
+  auto namingDecl = opaqueRoot->getDecl()->getNamingDecl();
+  auto module = namingDecl->getModuleContext();
+  if (module->isResilient() && module != modulePerformingSubstitution &&
+      !namingDecl->getAttrs().hasAttribute<InlinableAttr>()) {
```

Comment:
Please us getResilienceExpansion() instead of directly checking for an inlinable attribute; you're missing a number of cases this way